Src/GoogleApis.Auth/OAuth2/Requests/GoogleAuthorizationCodeRequestUrl.cs

```diff
@@ -58,6 +59,13 @@ public class GoogleAuthorizationCodeRequestUrl : AuthorizationCodeRequestUrl
         [Google.Apis.Util.RequestParameterAttribute("include_granted_scopes",
             Google.Apis.Util.RequestParameterType.Query)]
         public string IncludeGrantedScopes { get; set; }
+        
+        /// Gets or sets a collection of user defined query parameters to facilitate any not explicitly supported
+        /// by the library which will be included in the resultant authentication URL.
+        /// </summary>
+        [Google.Apis.Util.RequestParameterAttribute("user_defined_query_parameters",

```

Comment:
I think that the thing that I don't like here is that we have a name for the RequestParameterAttribute. People might wonder what is this user_defined_... parameter?

I suggest adding another constructor to https://github.com/google/google-api-dotnet-client/blob/1dce119eb5200be57a6ff24159b895594d6bd558/Src/Support/GoogleApis.Core/Apis/Util/RequestParameterAttribute.cs without name, just type and use it here. What do you say?